### PR TITLE
support numeric command

### LINKED ISSUE #36: [API Proposal] Numeric Commands
#### Syntax
```
    INCR cluster  key 
```
Increments the number stored at key by one. If the key does not exist, it is set to 0 before performing the operation.

Resp Reply:
Integer reply: the value of the key after the increment.

#### Syntax
```
    INCRBY cluster  key increment
```
Increments the number stored at key by the value of the parameter increment. If the key does not exist, it is set to 0 before performing the operation.

Resp Reply
Integer reply: the value of the key after the increment.

#### Syntax
```
    DECR cluster key
```

Decrements the number stored at key by one. If the key does not exist, it is set to 0 before performing the operation. An error is returned if the key contains a value of the wrong type or contains a string that can not be represented as integer.

RESP Reply
Integer reply: the value of the key after decrementing it.

#### Syntax
```
    DECRBY cluster  key decrement
```
Decrements the number stored at key by the value of parameter decrement. If the key does not exist, it is set to 0 before performing the operation. An error is returned if the key contains a value of the wrong type or contains a string that can not be represented as integer.

Resp Reply
Integer reply: the value of the key after decrementing it.

### PR DESCRIPTION
**Pull request by SaeedSafi1999**
as we talked is issue #36 
add INCR and DECR for application
add in persistence commands
add public api for them

### CODE REVIEW

File: src/cache/cache.rs [hunk 1]
```diff
@@ -1,23 +1,57 @@
...
 use std::io::{BufRead, BufReader};
 use std::path::PathBuf;
 use std::sync::{Arc, Mutex};
 use std::time::{Duration, Instant};
 
-use crate::memory_handling;
+#[derive(Clone, Serialize)]
+pub enum CacheType {
+    Str = 1,
+    Num = 2,
```
Comment (Alirexaa) [inline src/cache/cache.rs:20]:
```suggestion
    Int = 2,
```

File: src/cache/cache.rs [hunk 2]
```diff
@@ -1,23 +1,57 @@
...
+impl CacheType {
+    fn as_i32(&self) -> &i32 {
+        match self {
+            CacheType::Num => &1,
+            CacheType::Str => &2,
+        }
+    }
+    fn as_str(&self) -> &str {
+        match self {
+            CacheType::Num => "Number",
```
Comment (Alirexaa) [inline src/cache/cache.rs:32]:
```suggestion
            CacheType::Int => "integer",
```

File: src/cache/cache.rs [hunk 3]
```diff
@@ -1,23 +1,57 @@
...
+#[derive(Clone, Serialize)]
+pub enum CacheType {
+    Str = 1,
+    Num = 2,
+}
+
+impl CacheType {
+    fn as_i32(&self) -> &i32 {
+        match self {
+            CacheType::Num => &1,
```
Comment (Alirexaa) [inline src/cache/cache.rs]:
```suggestion
            CacheType::Int => &1,
```

File: src/cache/cache.rs (hunk 2 above)
Comment (Alirexaa) [inline src/cache/cache.rs]:
```suggestion
            CacheType::Int => "integer",
```

**Review by Alirexaa — COMMENTED** [review]
We should name this `Int` instead of number because we store value as `i32`

File: src/cache/cache.rs [hunk 4]
```diff
@@ -51,6 +85,175 @@ impl Cache {
         cache
     }
 
+    pub fn add_incr(
+        &mut self,
+        cluster: String,
+        key: String,
+        value: Option<Vec<u8>>,
```
Comment (Alirexaa) [inline src/cache/cache.rs]: Why the value type is `Option<Vec<u8>>`?
should be `Option<i32>`?

**Comment by SaeedSafi1999** [comment]
change all the things we talked 
add some helpers functions to code more easier 
vec_to_i32
i32_to_vec
CC @Alirexaa

**Comment by SaeedSafi1999** [comment]
I test it every thing just file as past.

**Review by Alirexaa — APPROVED** [review]

**Pull request merged**